ppmpy/ppm.py

```diff
@@ -809,6 +816,250 @@ def compute_Ur(self, fname, num_type='ndump'):
         Ur = np.sqrt(U**2 - Ut**2)
         return Ur
 
+    def compute_nabla_T(self, fname, num_type='ndump'):
+        '''
+        Returns the actual temperature gradient nabla
+        '''
+        
+        if self.__isyprofile:
+            T = self.get('T9', fname, num_type=num_type, resolution='l')
+            p = self.get('P', fname, num_type=num_type, resolution='l')
+
+        if self.__isRprofSet:
+            T = self.get('T9', fname, num_type=num_type, resolution='l')
+            p = self.get('P0', fname, num_type=num_type, resolution='l') + \
+                self.get('P1', fname, num_type=num_type, resolution='l')
+
+        nabla_T = cdiff(np.log(T))/(cdiff(np.log(p)) + 1e-100)
+        return nabla_T
+
+    def compute_nabla_T_ad(self, fname, num_type='ndump', radeos=True):
+        '''
+        Returns the adiabatic temperature gradient nabla_ad
+        '''
+
+        if radeos:
+            beta = self.compute_pgas_by_ptot(fname, num_type=num_type)
+            nabla_T_ad = (8-6*beta)/(32-24*beta-3*beta**2)
+        else:
+            if self.__isyprofile:
+                r = self.get('Y', fname, num_type=num_type, resolution='l')
+
+            if self.__isRprofSet:
+                r = self.get('R', fname, num_type=num_type, resolution='l')
+
+            nabla_T_ad = (2./5.)*np.ones(len(r))
+            
+        return nabla_T_ad
+
+
+    def compute_nabla_T_rad(self, fname, num_type='ndump'):
+        '''
+        Returns the radiative temperature gradient nabla_rad
+
+        Assumes a constant luminosity totallum for all radii
+        '''
+
+        totallum = self.get('totallum', fname, num_type=num_type, resolution='l')
+        self.__messenger.warning('PPMtools.compute_nabla_T_rad() assumes that all '
+                                 'the luminosity is injected at the center R=0')
+
+        m = self.compute_m(fname, num_type=num_type)
+        T = 1e9*self.get('T9', fname, num_type=num_type, resolution='l')
+        kappa = self.compute_kappa(fname)
+
+        if self.__isyprofile:
+            p = 1e19*self.get('P', fname, num_type=num_type, resolution='l')
+
+        if self.__isRprofSet:
+            p = 1e19*(self.get('P0', fname, num_type=num_type, resolution='l') + \
+                self.get('P1', fname, num_type=num_type, resolution='l'))
+            
+        # compute nabla_T_rad with all quantities in cgs units
+        arad = 7.5646e-15
+        cc = 2.99792458e10
+        Gconst = 6.67430e-8
+        totallum = totallum*1.e43
+        m = m*1.e27
+        nabla_T_rad = 3*kappa*totallum*p / (16*np.pi*arad*cc*Gconst*m*T**4+1e-100)
+            
+        return nabla_T_rad
+
+    def compute_lum_conv(self, fname, num_type='ndump'):
+        '''
+        Returns the convective luminosity (Fconv*4piR^2)
+        
+        Includes both the kinetic energy term and the enthalpy term
```

Comment:
Huaqing told me that rhoHrH includes both the enthalpy term and the kinetic term (despite the name of the variable suggesting otherwise), which Paul and him have been calling the convective flux (we can always argue whether this is the proper terminology). I will check again with Huaqing.

Reply:
It is correct as it is:
![image](https://user-images.githubusercontent.com/15151278/161595611-9e147e42-7255-42c3-a387-0e07487886da.png)

Reply:
I think this is now resolved.

Reply:
Let's add a statement to the doc string that this is something that may require clarification. Otherwise, ok.